jcli/src/jcli_lib/certificate/sign.rs

```diff
@@ -244,3 +252,43 @@ where
     let sig = PoolOwnersSigned { signatures: sigs };
     Ok(to_signed_certificate(payload, sig))
 }
+
+pub(crate) fn update_proposal_sign<P: Payload>(
+    update_proposal: UpdateProposal,
+    keys_str: &[String],
+    builder: TxBuilderState<SetAuthData<P>>,
+) -> Result<SignedCertificate, Error> {
+    if keys_str.len() > 1 {
+        return Err(Error::ExpectingOnlyOneSigningKey {
+            got: keys_str.len(),
+        });
+    }
+
+    let private_key = parse_ed25519_secret_key(keys_str[0].trim())?;
```

Comment:
And while we are on this topic, `keys_str[0]` will panic on an empty slice and this is not checked earlier. The proper way to handle this would be something like

```rust
match keys {
    [key] => Ok(key),
    keys => Err(Error::ExpectingOnlyOneSigningKey { got: keys.len() }),
}
```